htdocs/core/class/commonobject.class.php

```diff
@@ -6767,7 +6767,7 @@ public function showOptionals($extrafields, $mode = 'view', $params = null, $key
 					// Convert float submited string into real php numeric (value in memory must be a php numeric)
 					if (in_array($extrafields->attributes[$this->table_element]['type'][$key], array('price', 'double')))
 					{
-						$value = (!empty($value)) ? price2num($value) : $this->array_options['options_'.$key];
+						$value = (!empty($value) || GETPOSTISSET($keyprefix.'options_'.$key.$keysuffix)) ? price2num($value) : $this->array_options['options_'.$key];
```

Comment:
@simicar29 
it is not useful to do this check! 
it already performed on the assignment of the value of the variable "$value"

Reply:
Hi @hregis, I am afraid it is required. $value gets assigned when posted or when a default value was defined. Alternative could be to move up the check on type where value gets assigned to directly apply the price2num conversion. What do you think ?

Reply:
Old code looks good.
When we submit, if we submit a value (so GETPOSTISSET is true), we use it, whatever is value.
If nothing was submitted, we keep using $this->array_options['options_'.$key];

Reply:
Well, with old code you never get default values here since they are used when GETPOSTISSET is false.